instance/models/openedx_instance.py

```diff
@@ -449,24 +451,27 @@ def require_user_creation_success(self):
     def terminate_obsolete_appservers(self, days=2):
         """
         Terminate app servers that were created more than `days` before now, except:
-        - the active appserver if there is one,
+        - the active appserver(s) if there are any,
         - a release candidate (rc) appserver, to allow testing before the next appserver activation
           (we keep the most recent running appserver)
         - a fallback appserver, for `days` after activating an appserver, to allow reverts
-          (we keep the most recent running appserver created before the current activation)
+          (we keep the most recent running appserver created before the latest activation)
         """
-        active_appserver = self.active_appserver
+        latest_active_appserver = None
+        if self.get_active_appservers().count():
+            latest_active_appserver = self.get_active_appservers().latest('last_activated')
         fallback_appserver = None
         rc_appserver = None
         now = timezone.now()
 
         for appserver in self.appserver_set.all().order_by('-created'):
-            if appserver == active_appserver:
+            # Skip active appservers
+            if appserver.is_active:
                 continue
 
-            # Keep a running appserver as fallback for `days` after activation, to allow reverts
-            if active_appserver and appserver.created < active_appserver.last_activated:
-                if not sufficient_time_passed(active_appserver.last_activated, now, days) \
+            # Keep a running appserver as fallback for `days` after latest activation, to allow reverts
```

Comment:
Don't we need to keep the previous set of appservers to allow reverts, rather than just one? Otherwise we'll end up in a situation where we can't revert, because there would only be one of the previous apservers left running, which couldn't be enough to handle the traffic and/or ensure the level of fault tolerance we are after.

Same remark about the rc server -- we'll probably want to provision multiple ones, and switch them all at once.

It sounds like appservers need to be grouped by release/version, ie we are missing a logical layer. This specific logic shouldn't have to change imho, and simply apply to a set of appservers meant to be activated together, rather to individual appservers.